processing/src/main/java/io/druid/query/groupby/epinephelinae/ConcurrentGrouper.java

```diff
@@ -94,6 +96,8 @@ public ConcurrentGrouper(
           )
       );
     }
+
+    this.entryComparator = keySerdeFactory.factorize().entryComparator();
```

Comment:
@himanshug that works for me, it reflects the fact that the buffer version of a key can change as the keySerde resets, but the object version is always consistent.

I still also think it'd be nice for it to be `Comparator<T> objectComparator()` rather than `Comparator<Entry<T>>`.